pkg/ebpf/c/tracee.bpf.c

```diff
@@ -4728,6 +4729,27 @@ int BPF_KPROBE(trace_security_inode_mknod)
     return events_perf_submit(&data, SECURITY_INODE_MKNOD, 0);
 }
 
+SEC("kprobe/__register_chrdev ")
+int BPF_KPROBE(trace___register_chrdev)
+{
+    event_data_t data = {};
+    if (!init_event_data(&data, ctx))
+        return 0;
+
+    if (!should_trace(&data.context))
+        return 0;
+
+    unsigned int major_number = PT_REGS_PARM1(ctx);
```

Comment:
This is problematic as @yanivagman said because the major might be received by the return value of the function.
In my opinion, you can make this program a `kretprobe` and than analyze the return value as the major in the case `major==0`.